### ISSUE TITLE
Amazon reviews dataset

### ISSUE THREAD

**Issue by alexey-milovidov** [ClickHouse/ClickHouse#30627]
Found here: https://habr.com/en/post/512084/

Description: https://s3.amazonaws.com/amazon-reviews-pds/readme.html

License:

> In addition to the license rights granted under the Conditions of Use, Amazon or its content providers grant you a limited, non-exclusive, non-transferable, non-sublicensable, revocable license to access and use the Reviews Library for purposes of academic research. You may not resell, **republish**, or make any commercial use of the Reviews Library or its contents, including use of the Reviews Library for commercial research, such as research related to a funding or consultancy contract, internship, or other relationship in which the results are provided for a fee or **delivered to a for-profit organization**. You may not (a) link or associate content in the Reviews Library with any personal information (including Amazon customer accounts), or (b) attempt to determine the identity of the author of any content in the Reviews Library. If you violate any of the foregoing conditions, your license to access and use the Reviews Library will automatically terminate without prejudice to any of the other rights or remedies Amazon may have.

TLDR: I can do research, but we cannot make it accessible through our Playground or publish in our docs.

**Comment by alexey-milovidov** [comment]
Partially blocked by #36275.

**Comment by rfraposa** [comment]
> The format is non-standard: it's like TSV but without backslash escaping.
So, backslash can appear at the end of String value.

This makes it hard to include in our docs. Someone would have to download the files and the insert them into ClickHouse, because unfortunately the following query does not work. I think we should only include datasets in the docs that can be directly inserted into a ClickHouse server (unless it's a small dataset).

```
DESCRIBE TABLE s3('https://s3.amazonaws.com/amazon-reviews-pds/tsv/amazon_reviews_us_Wireless_v1_00.tsv.gz', 'TSVWithNames')

0 rows in set. Elapsed: 1.304 sec.

Received exception from server (version 23.3.1):
Code: 117. DB::Exception: Received from localhost:9000. 
DB::Exception: Rows have different amount of values: Cannot extract table structure from TSVWithNames format file.
You can specify the structure manually. (INCORRECT_DATA)
```

FWIW: the data is also in a Parquet format, but I can not figure out the URLs to download the Parquet files

**Comment by alexey-milovidov** [comment]
If a dataset cannot be inserted, it is our problem.
We should show in the docs how to work with a wide variety of datasets.

**Comment by alexey-milovidov** [comment]
PS. We can describe the usage of the dataset in our docs, it is not forbidden by the license.

**Comment by alexey-milovidov** [comment]
@rfraposa the comment suggests utilizing the `input_format_allow_errors_ratio`, it is a good, practical example.

**Comment by rfraposa** [comment]
The "allow errors" didn't work, but it led me to spend more time on this and I found a simple way to get the data ingested in ClickHouse Cloud. PR is here - https://github.com/ClickHouse/ClickHouse/pull/48129.  

I'm waiting for my Production instance to have `allow_experimental_inverted_index` enabled so I can build an inverted index and see how well that improves some of the queries.

**Comment by alexey-milovidov** [comment]
Here is a mistake:

```
* REPLACE(vine = 'Y' AS vine, verified_purchase = 'Y' AS verified_purchase)
```

like you want to replace them with Bool, but you left the columns in the schema as FixedString.

**Comment by alexey-milovidov** [comment]
> toYYYYMM(review_date)

Ok, but if you use `toStartOfMonth`, the result is kept in the `Date` data type, it can be better for further data processing (e.g., charts).

**Comment by alexey-milovidov** [comment]
I really like the tutorial and the creative approach to data exploration!

**Comment by rfraposa** [comment]
> Here is a mistake:
> 
> ```
> * REPLACE(vine = 'Y' AS vine, verified_purchase = 'Y' AS verified_purchase)
> ```


That is a mistake = thank you for noticing. I used the following command to insert the data - I'll update the doc:

```
INSERT INTO amazon_reviews
SELECT 
   * 
FROM s3Cluster(
    'default',
    'https://s3.amazonaws.com/amazon-reviews-pds/tsv/amazon_reviews_us_*.tsv.gz', 
    'TSVWithNames',
    'review_date Date,
    marketplace LowCardinality(String),
    customer_id UInt64,
    review_id String,
    product_id String,
    product_parent UInt64,
    product_title String,
    product_category LowCardinality(String),
    star_rating UInt8,
    helpful_votes UInt32,
    total_votes UInt32,
    vine FixedString(1),
    verified_purchase FixedString(1),
    review_headline String,
    review_body String'
    )
```

**Comment by rfraposa** [comment]
I updated the `toStartOfMonth` advise also - thanks! It's here: https://github.com/ClickHouse/ClickHouse/pull/48174

**Comment by alexey-milovidov** [comment]
Amazon removed the access to this dataset :(

**Comment by rfraposa** [comment]
This is sad news. I use the dataset in many examples in the training, and that content will now likely need to be updated. Lesson learned on my part...in the future I will make sure we have the data in our own S3 buckets before using it heavily in training examples.
